### DISCUSSION TITLE
How to operate fuzzing tests?

### DISCUSSION THREAD

**Discussion by gordontcp** [fwupd/fwupd#5297] [Q&A] [answered]
I built the code according to `https://github.com/fwupd/fwupd` instructions,
```
CC=hfuzz-clang meson --default-library=static \
-Dudevdir=/tmp -Dsystemd_root_prefix=/tmp \
-Dplugin_redfish=disabled -Dcurl=disabled \
-Dintrospection=false ../
ninja install
ninja fuzz-firmware
ninja fuzz-tpm-eventlog
```
The last two lines get the following error message:
```
ninja: error: unknown target 'fuzz-firmware'
ninja: error: unknown target 'fuzz-tpm-eventlog'
```
My system is as follows:
```
Linux fedora 6.0.9-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Nov 16 17:50:45 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
[x@fedora build]$ fwupdmgr --version
runtime org.freedesktop.fwupd 1.8.6
runtime org.freedesktop.fwupd-efi 1.3
compile org.freedesktop.gusb 0.3.10
runtime com.hughsie.libjcat 0.1.12
runtime org.kernel 6.0.9-200.fc36.x86_64
runtime com.dell.libsmbios 2.4
compile com.hughsie.libjcat 0.1.12
compile org.freedesktop.fwupd 1.8.6
runtime org.freedesktop.gusb 0.3.10
```
My question is as follows:
1. How to build for ‘ninja fuzz-firmware’, ‘ninja fuzz-tpm-eventlog’?
2. Is there any detailed instructions on how to operate the fuzzing tests?
Any suggestions are greatly appreciated!
Thanks!

**Comment by hughsie** [comment, marked as answer]
Ahh, those instructions are pretty wrong now, sorry about that. We've pretty heavily leant-into using oss-fuzz as that scales much better than just letting my laptop run overnight :) If you see contrib/ci/oss-fuzz.py in the code checkout, we actually compile some deps and build dozens of tiny fuzzing targets. Does that help at all?

**Reply by gordontcp** [reply]
Many thanks!
